controllers/events.js

```diff
@@ -172,27 +172,28 @@ function saveEvent(request, response){
 function eventDetail (request, response) {
   var ev = events.getById(parseInt(request.params.id));
   if (ev === null) {
-    response.status(404).send('No such event');
+    return response.status(404).send('No such event');
   }
   response.render('event-detail.html', {event: ev});
 }
 
 function rsvp (request, response){
   var ev = events.getById(parseInt(request.params.id));
+  var contextData = {errors: [], event: ev};
+
   if (ev === null) {
-    response.status(404).send('No such event');
+    return response.status(404).send('No such event');
   }
-
-  if(validator.isEmail(request.body.email)){
-    ev.attending.push(request.body.email);
-    response.redirect('/events/' + ev.id);
-  }else{
-    var contextData = {errors: [], event: ev};
+  if(!validator.isEmail(request.body.email) || request.body.email.toLowerCase().indexOf('@yale.edu') === -1){
     contextData.errors.push('Invalid email');
     response.render('event-detail.html', contextData);    
+  }else{
+    ev.attending.push(request.body.email.toLowerCase());
```

Comment:
While this is a good idea, I think it causes the (poorly written) test not to pass.